libpod/image/image.go

```diff
@@ -609,7 +609,7 @@ func (i *Image) Layer() (*storage.Layer, error) {
 }
 
 // History gets the history of an image and information about its layers
-func (i *Image) History(ctx context.Context) ([]ociv1.History, []types.BlobInfo, error) {
+func (i *Image) History(ctx context.Context, imageIDs *[]string) ([]ociv1.History, []types.BlobInfo, error) {
```

Comment:
Why are we passing in `imageIDs` here? Can't we compute this from the image's own ID?

Reply:
It seems like we should be returning imageIDs instead. Also, we should probably just return an array of one struct, containing the `ociv1.History`, `types.BlobInfo`, and image ID for the layer in history, so we don't have to worry about potentially mismatched array sizes.

Reply:
I want to pass in an array that I can modify, so that is why I was doing that. But you are right I can just return it instead. I am taking care of the size mismatch in cmd/podman/history.go.
Also we need to consider the order `ocivi.History` and `types.BlobInfo` are returned in. I think they are opposite of each other and this is also taken care of in cmd/podman/history.go.
Changing it one struct here will add more complexity, but if that is preferred I can make the change.

Reply:
I think that returning one struct is probably better than returning sets of arrays that are potentially in reverse order... That seems really not-obvious to work with.

As a bonus, it might let us remove code duplication with varlink, which is always good.

Sorry for the extra work!

Reply:
sure, no worries! Working on it.